### ISSUE TITLE
[BUG]: `import a.b` made `b` available but not `a.b`

### ISSUE THREAD

**Issue by soraros** [modularml/mojo#1939]
### Bug description

Related to #1415. Python doesn't do this, and it's super annoying.

### Steps to reproduce

```mojo
import algorithm.swap

fn swap():
  ...

fn main():
  algorithm.swap.swap()
```

### System information

```shell
Mojo 24.1 on Docker, Intel Mac
```

**Comment by soraros** [comment]
CC @frasercrmck. I don't think the following is correct, for we should only be able to access the `math` module in fully qualified form. 

```mojo
# Mojo 0.26.2.0.dev2026030405 (1cf174b8)
import std.math

fn math():  # error: invalid redefinition of 'math'
  ...

fn main():
  ...
```

**Comment by frasercrmck** [comment]
@soraros agreed, that's incorrect. I've just landed a change which makes the current behaviour deprecated with a warning for the next release. We'll make it a hard error in the release after that.

**Comment by frasercrmck** [comment]
Closing as I've merged the hard-error change. This change should propagate to the nightlies in the next day or two.
